### ISSUE TITLE
feat(ml-spec): Add ConditionalAttributeType to Attribute.type union

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3685]
## Summary

Add `ConditionalAttributeType` to the `Attribute` interface's `type` union in `@markuplint/ml-spec`.

This is the last chance to change the `Attribute` interface before v5.0 (semver constraint). **v5.0 ships the type definition only.** Validation logic and spec data updates will follow in subsequent issues.

## Background

The current `Attribute.type` is `AttributeType | AttributeType[]` — attribute types are statically determined. There is no way to express cases where an attribute's type depends on the value of another attribute (conditional type switching).

### Use cases

- **#3598**: `input[type=color]` value should be `#rrggbb`, `input[type=url]` value should be a URL — the `value` type depends on the `type` attribute (~97 missed errors)
- **#3189**: `link[rel=preload]` `as` and `link[rel=modulepreload]` `as` allow different enum values

### New type definition

```typescript
type ConditionalAttributeType = {
  readonly condition: AttributeCondition; // CSS selector
  readonly type: AttributeType | [AttributeType, ...AttributeType[]];
};
```

### Extend `Attribute.type`

```typescript
// Before
type: AttributeType | readonly AttributeType[];

// After
type: AttributeType | readonly AttributeType[] | readonly ConditionalAttributeType[];
```

### JSON Schema update

Add `ConditionalAttributeType` definition to `/packages/@markuplint/ml-spec/schemas/attributes.schema.json`.

## Scope for v5.0

- [ ] Add `ConditionalAttributeType` type definition
- [ ] Extend `Attribute.type` union
- [ ] Add definition to JSON Schema
- [ ] Update type generation

## Out of scope (follow-up issues)

- Validation logic (conditional resolution in `isValidAttr`) → #3598
- Spec data application (`spec.input.jsonc`, etc.) → #3598, #3189

## Related

- #3684 — v5.0 release requirements (Group B)
- #3598 — input value validation (follow-up: logic implementation)
- #3189 — condition-specific enum values (follow-up: logic implementation)
